modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -293,7 +227,11 @@ private void replace(String format, StringBuffer result, MessageContext synCtx)
         String replacementValue = null;
         Matcher matcher;
 
-        matcher = pattern.matcher(format);
+        if (mediaType != null && (mediaType.equals(JSON_TYPE) || mediaType.equals(TEXT_TYPE))) {
```

Comment:
Can't we change this to 
`JSON_TYPE.equals(mediaType) || TEXT_TYPE.equals(mediaType) `
and avoid the null check?

Reply:
+1